### DISCUSSION TITLE
Defining a clone-component

### DISCUSSION THREAD

**Discussion by tomasz-zwak** [GrapesJS/grapesjs#5634] [Q&A]
First of all I have to say I am very impressed at what the library can do! Absolutely amazing work.

I have a question regarding defining custom components, maybe some of you guys can show me the right path.

I need to create a component that, once placed on the canvas and edited, will clone it's contents into every other component of the same type. Imagine Microsoft Word footers/headers type of functionality.

Was thinking of listening to editor:update event, then, if the update happened in such component, copy it's contents and replace the contents of all the others with it. 

Maybe there is some better way?

**Comment by artf** [comment]
Hi @tomasz-zwak here you can find the discussion about [Symbols](https://github.com/GrapesJS/grapesjs/discussions/4317)

**Reply by tomasz-zwak** [reply]
Very interesting. Thank you @artf !
